src/target.c

```diff
@@ -108,61 +111,107 @@ stumpless_add_entry( struct stumpless_target *target,
   }
 
   clear_error(  );
+ 
+  if( stumpless_get_option( target, STUMPLESS_OPTION_PERROR ) == STUMPLESS_OPTION_PERROR ){
+	// setting is_log_formatted true concludes that STUMPLESS_OPTION_PERROR option is
+	// set, thus only checking this flag is enough for further checks
+	// NOTE : Calling the format_entry twice because if the buffer is for target who
+	// require unformatted entry would need not to call format_entry.
+	is_log_formatted = true;
```

Comment:
this boolean is unnecessary - you can simply check for whether the buffer is NULL further down in the function, rather than this separate flag